### PR TITLE
Feature/resource barrier

### PR DESCRIPTION
**Pull request by LeonBrands**
Adds CommandList::resourceBarrier(). Allows for applications to insert memory dependencies during command list execution. Transition barriers can be used to change the perceived usage of resources, whereas ReadWrite barriers can be inserted in-between shader operations that may perform read/write operations on resources manually.

Also fixes a couple of small validation bugs that I ran into in the meantime.

### CODE REVIEW

File: legion/engine/llri-vk/detail/device.cpp [hunk 1]
```diff
@@ -239,41 +241,94 @@ namespace llri
...
-                static_cast<VolkDeviceTable*>(m_functionTable)->vkDestroyBuffer(static_cast<VkDevice>(m_ptr), buffer, nullptr);
+                table->vkDestroyBuffer(static_cast<VkDevice>(m_ptr), buffer, nullptr);
 
-            static_cast<VolkDeviceTable*>(m_functionTable)->vkFreeMemory(static_cast<VkDevice>(m_ptr), memory, nullptr);
+            table->vkFreeMemory(static_cast<VkDevice>(m_ptr), memory, nullptr);
 
             return internal::mapVkResult(r);
         }
+        
+        // this part is necessary because in vulkan images are created int he UNDEFINED layout
```
Comment (GlynLeine) [inline legion/engine/llri-vk/detail/device.cpp]: "int he" type for "in the" i assume

File: legion/engine/llri-vk/detail/device.cpp [hunk 2]
```diff
@@ -239,41 +241,94 @@ namespace llri
...
+            
+            // Record commandbuffer to transition texture from undefined
+            VkCommandBufferBeginInfo beginInfo {};
+            beginInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
+            beginInfo.pNext = nullptr;
+            beginInfo.flags = VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT;
+            beginInfo.pInheritanceInfo = nullptr;
+            table->vkBeginCommandBuffer(static_cast<VkCommandBuffer>(m_workCmdList), &beginInfo);
+            
+            // use the texture format to detect the aspect flags
```
Comment (GlynLeine) [inline legion/engine/llri-vk/detail/device.cpp:285]: You're a little inconsistent in whether you start comments with upper case letters or lower case. whether that's an issue i'll leave up to you

Comment (LeonBrands) [reply legion/engine/llri-vk/detail/device.cpp:285]: I suppose that's a bit of a concern, I'm not sure what my preference is for comments

Comment (LeonBrands) [reply legion/engine/llri-vk/detail/device.cpp:285]: I think I'm going to leave this to a future PR where I'll establish a standard for API & implementation comments and make it consistent throughout

**Review by GlynLeine — APPROVED** [review]

**Pull request merged**